<?xml version="1.0"?>
<launch>
    <node pkg="yf_manager" name="MissionXYZ_node" type="MissionXYZ_node" output="screen">
        <remap from="/odom" to="/mavros/local_position/odom"/>
        <param name="mission/sendOneByOne" value="true" />
        <param name="mission/wps_threshold" value="2.0" />
        <param name="mission/control_mode" value="3" />
        <param name="mission/handle_wpts_xy" value="UseOffboardPoint" />
        <param name="mission/handle_wpts_z" value="UseOffboardHeight" />
        <param name="mission/yf_mission_log_filename" value="$(find yf_manager)/yf_mission.txt" />

        <param name="mission/waypoint_num" value="1" />

        <param name="mission/waypoint0_x" value="35.0" />
        <param name="mission/waypoint0_y" value="0.0" />
        <param name="mission/waypoint0_z" value="0.0" />
        <param name="mission/waypoint0_vm" value="3.0" />
        <param name="mission/waypoint0_am" value="2.0" />

        <param name="mission/waypoint1_x" value="20.0" />
        <param name="mission/waypoint1_y" value="20.0" />
        <param name="mission/waypoint1_z" value="2.0" />
        <param name="mission/waypoint1_vm" value="2.0" />
        <param name="mission/waypoint1_am" value="2.0" />

        <param name="mission/waypoint2_x" value="0.0" />
        <param name="mission/waypoint2_y" value="20.0" />
        <param name="mission/waypoint2_z" value="2.0" />
        <param name="mission/waypoint2_vm" value="2.0" />
        <param name="mission/waypoint2_am" value="2.0" />

        <param name="mission/waypoint3_x" value="0.0" />
        <param name="mission/waypoint3_y" value="0.0" />
        <param name="mission/waypoint3_z" value="2.0" />
        <param name="mission/waypoint3_vm" value="2.0" />
        <param name="mission/waypoint3_am" value="2.0" />
    </node>

    <node pkg="yf_manager" name="PlanFSM_node" type="PlanFSM_node" output="screen">
        <remap from="/odom" to="/mavros/local_position/odom"/>
        <remap from="/depth" to="/realsense_plugin/camera/depth/image_raw"/>

        <param name="paramfile/path" value="$(find yf_manager)/config/gz.yaml" type="string"/>

        <param name="fsm/control_point_distance" value='0.4'/>
        <!-- 地图的边界 -->
        <param name="fsm/planning_horizon" value='15.0'/>
        <!-- 到达终点附近，不再规划 -->
        <param name="fsm/no_replan_thresh" value='3.0'/>
        <!-- 每段轨迹至少执行的距离 -->
        <param name="fsm/replan_thresh" value='2.0'/>
        <param name="fsm/collsion_check_dist" value='1.0'/>
    </node>

</launch>
